Improve documentation with React components

pepibumur opened the pull request:
### Short description 📝
I found the documentation a bit hard to navigate. This PR uses `mdx` to represent each of the manifest models with a table. Something like this:

![image](https://user-images.githubusercontent.com/663605/58155720-bcf90e00-7c74-11e9-9c73-0acbce3dd55c.png)

### Solution 📦

The models are now represented with React components:

```js
<PropertiesTable props={[
{
  name: "Name",
  description: "The name of the target. The Xcode project target and the derivated product take the 
  same name.",
  type: "String",
  optional: false,
  default: ""
}]}/>
```

kwridan commented on docs/usage/2-manifest.mdx:
````diff
@@ -98,47 +99,238 @@ default: "[]"
+                    dependencies: [])
+```
+
+>>>>>>> Move more models to tables
````
yikes looks like the conflict markers were accidentally added

kwridan commented on docs/usage/2-manifest.mdx:
```diff
@@ -98,47 +99,238 @@ default: "[]"
-- **Environment (optional):** It's a `[String: String]` to defined variables that will be set to the scheme that Tuist automatically generates for the target.
+<Message info title="Relative paths" description="Either the tool or the path need to be defined. Moreover, Tuist verifies whether the launch path is valid and fail otherwise."/>
 
 ### Scheme
```
Is the idea to incrementally adopt the table for the various sections? e.g. Scheme still uses the list style

pepibumur replied:
I missed this one. It should be ported now to the new format 👍

pedropinerashopify commented:
@kwridan I [opened an issue](https://github.com/pedronauck/docz/issues/896) reporting the issue with the button on top of the document title when it's opened on mobile.

Regarding the layout on mobile, I was looking at the behaviour of the Table component that we are using from this [library](https://react.semantic-ui.com/collections/table/), and that's the default behaviour, which I agree is a bit odd. I can disable it but then the table would expand beyond the device viewport. Optionally, what I can do is presenting the attributes as a list if it's opened from a mobile device. What do you think @kwridan?

pepibumur commented:
@kwridan added some improvements:

* I improved the responsiveness of the tables when they are shown on a mobile device.

![image](https://user-images.githubusercontent.com/663605/58374020-13956f00-7f37-11e9-8afb-b8b1759d0f96.png)

* Changed the style of the code blocks. Docz, the library that we are using for the documentation generation uses [CodeMirror](https://codemirror.net/demo/theme.html) as a code highlighter, which unfortunately can't parse the Swift code accurately. I tried to get as close as possible to the Xcode highlighting but it's a bit far from ideal IMO. Still better than the dark that we had before though.
* Changed the default font to use the ones that Apple is using on the Swift website. I think with that font the website looks fresher and certainly more readable.
 
I also ported the scheme's documentation to the new format using components. Would you mind having another look before we merge it?

kwridan commented:
Ah it's a docz issue https://github.com/pedronauck/docz/issues/850

pepibumur commented:
I fixed both, the headers and the scrolling issues. Moreover, I enabled a button at the top-right corner to edit the page on GitHub:

![image](https://user-images.githubusercontent.com/663605/58402010-9b769880-805f-11e9-8604-0c577f914983.png)
